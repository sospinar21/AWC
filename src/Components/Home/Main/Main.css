.suggestions-menu {
  display: flex;
  width: 53%;
  margin: 4vh auto;
  justify-content: space-between;
  background-color: #dcdcdc57; }

.box {
  background-color: #fffffffc;
  box-shadow: rgba(0, 0, 0, 0.06) 0px 2px 4px 0px;
  width: 100%;
  border-width: 1px;
  border-style: solid;
  border-color: rgba(0, 0, 0, 0.1);
  border-image: initial;
  border-radius: 3px;
  padding: 0px;
  overflow: hidden; }

.news-feed {
  width: 100%;
  height: 99vh;
  display: flex;
  background-color: #f3dfdf1f; }

.side-container {
  width: 25%; }

.forum {
  margin-left: 14vw;
  width: 80%;
  height: 99vh;
  background-color: white;
  -webkit-box-shadow: 3px 3px 5px lightgrey;
  box-shadow: 0px 2px 8px lightgrey; }

.events-small {
  overflow-x: hidden;
  height: 50vh;
  overflow: scroll;
  padding: 0vh 2vh;
  background-color: white;
  margin: 1.3vh 1.3vh 1.3vh 0;
  box-sizing: border-box;
  box-shadow: 3px 3px 5px lightgrey; }
  .events-small h3 {
    color: #232b37; }

.titleFix {
  padding: 1.4vh;
  margin: 0vh 1.23vh 1.23vh 0;
  height: 5.5vh;
  background-color: #232b37;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
  line-height: 1.4;
  box-shadow: 3px 3px 5px lightgrey; }
  .titleFix a {
    color: white;
    font-weight: 200; }
  .titleFix a:hover {
    color: #f19db0;
    cursor: pointer; }

.studios-small {
  overflow-x: hidden;
  background-color: white;
  margin: 1.3vh 1.3vh 1.3vh 0;
  height: 34vh;
  padding: 0 2vh;
  box-sizing: border-box;
  box-shadow: 3px 3px 5px lightgrey; }
  .studios-small h3 {
    color: #232b37; }

.videos-small {
  background-color: #ffffff;
  overflow-x: scroll;
  width: 90%;
  margin-left: 10%;
  -webkit-box-shadow: 3px 3px 5px lightgrey;
  box-shadow: 2px 0px 8px lightgrey; }

.location {
  display: flex;
  width: 100%;
  height: 5.5vh;
  padding: 1vh 0 1vh 5%;
  box-sizing: border-box;
  text-align: center;
  background-color: #232b37;
  box-shadow: 3px 3px 5px lightgrey; }
  .location input {
    border: none;
    padding: 10px 15px;
    font-size: 1.6vh;
    width: 50%;
    color: black;
    outline: none;
    line-height: 1.5; }
  .location .submit-location {
    color: white;
    background-color: #b62d4d;
    width: 20%;
    padding: 0.8vh;
    font-family: 'Roboto', sans-serif;
    font-size: 1.6vh; }
  .location a {
    color: white;
    font-weight: 200; }
  .location a:hover {
    color: #f19db0;
    cursor: pointer; }

.navBar-forum {
  width: 75%;
  display: flex; }

.btns {
  justify-content: center;
  display: -ms-flexbox;
  display: flex;
  margin-right: 2vh;
  width: 30%; }

.location-form {
  width: 60%;
  display: flex; }

.music {
  width: 100%;
  height: 100%;
  background-image: url("https://www.sketchappsources.com/resources/source-image/music-player_1x.png"); }
