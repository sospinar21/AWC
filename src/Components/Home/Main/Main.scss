$flex : flex;
$shadow: -2px -2px 3px lightgrey;

.suggestions-menu {
  display: flex;
  width: 53%;
  margin: 4vh auto;
  justify-content: space-between;
  background-color: #dcdcdc57;
}


.box {
  background-color: #fffffffc;
  box-shadow: rgba(0, 0, 0, 0.06) 0px 2px 4px 0px;
  width: 100%;
  border-width: 1px;
  border-style: solid;
  border-color: rgba(0, 0, 0, 0.1);
  border-image: initial;
  border-radius: 3px;
  padding: 0px;
  overflow: hidden;
}

.news-feed {
  width: 100%;
  height: 78vh;
  display: flex;
  background-color: #8080801f;
}

.side-container {
  width: 25%;
  
}

.forum {
  width: 100%;
  height: 76vh;
  margin: 0 18px;
  background-color: white;
  -webkit-box-shadow: 3px 3px 5px lightgrey;
  box-shadow: 0px 2px 8px lightgrey;
}

.events-small {
  overflow-x: hidden;
  height: 35vh;
  overflow: scroll;
  padding: 0vh 2vh;
  background-color: white;
  margin: 1.3vh 1.3vh 1.3vh 0;
  box-sizing: border-box;
  box-shadow: 3px 3px 5px lightgrey;

  h3{
    color:#232b37;
  }
}

.titleFix {
  padding: 1.4vh;
  margin: 1vh 1.23vh 1.23vh 0;
  height: 5vh;
  background-color: #232b37;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;

  h3 {
    color: white;
  }
}


.studios-small {
  overflow-x: hidden;
  background-color: white;  
  margin: 1.3vh 1.3vh 1.3vh 0;
  height: 26vh;
  padding: 0 2vh;
  box-sizing: border-box;
  box-shadow: 3px 3px 5px lightgrey;

  h3{
    color:#232b37;
  }
}

.videos-small {
  background-color: #ffffff;
  overflow-x: scroll;
  width: 90%;
  margin-left: 10%;
  -webkit-box-shadow: 3px 3px 5px lightgrey;
  box-shadow: 2px 0px 8px lightgrey;     
}

.location {
  display: flex;
  width: 100%;
  height: 7vh;
  padding: 1vh 0 1vh 5%;
  box-sizing: border-box;
  text-align: center;
  background-color: #232b37;
  box-shadow: 3px 3px 5px lightgrey;

  input {
    font-size: 2vh;
    border: 1px solid #dcdcdc;
    margin: auto;
    padding: 0.7vh;
    width: 30%;
    margin-left: 0vh;
    font-family: 'Poiret One', cursive;
    background-color: #922e48;

    ::placeholder {
      color: white;
    }
}

a {
  color: rgb(255, 255, 255);
}

a:hover {
  color: black;
  cursor: pointer;
  background-color: transparent;
}
}

.active {
  border-bottom:2px solid pink;
}

.navBar-forum {
  width: 75%;
  display: flex;
}

.btns{
  justify-content: center;
    display: -ms-flexbox;
    display: flex;
    margin-right: 2vh;
    width: 30%;
}