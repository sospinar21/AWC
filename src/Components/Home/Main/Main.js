import React, { Component } from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import Events from '../Events/Events';
import NavBar from '../../Home/NavBar/NavBar';
import Studios from '../Studios/Studios';
import './Main.css';
import Videos from '../Videos/Videos';
import Community from '../Community/Community';
import ApiCalls from '../../../Helper/ApiCalls/ApiCalls';
import { addLocation, addEvents, addStudios } from '../../../Actions/actions';
import  SelectedEvent  from '../SelectedEvent/SelectedEvent';
import { userSignout } from '../../../Helper/Users/Users';

export class Main extends Component {
  constructor() {
    super();

    this.state = {
      locations: [],
      selectedLocation:'',
      communityActive:true,
      studiosActive: false,
      eventsActive:false
    };
  }

updateState = (e) => {
  const userInput = e.target.value;
  this.setState({selectedLocation:userInput});
  this.fetchLocation(userInput);
}

fetchLocation = async (userInput) => {
  const api = new ApiCalls();
  const location = userInput;
  const suggestions = await api.fetchGoogle(location);
  await this.displaySuggestions(suggestions);
    
}

displaySuggestions = (suggestions) => {
  if (suggestions){  
    this.setState({locations:suggestions});
  } 
}

selectedLocation = async (e) => {
  e.preventDefault();
  const api = new ApiCalls();
  this.props.addLocation(this.state.selectedLocation);
  const suggestedEvents = await api.fetchEvents(this.state.selectedLocation);
  const suggestedStudios = await api.fetchStudios(this.state.selectedLocation);
  this.props.addEvents(suggestedEvents);
  this.props.addStudios(suggestedStudios);
}

makeStudioActive = () => {
  this.setState({
    communityActive: false,
    studiosActive: true,
    eventsActive:false
  });
}

makeCommunityActive = () => {
  this.setState({
    communityActive: true,
    studiosActive: false,
    eventsActive:false
  });
}

makeEventsActive = () => {
  this.setState({
    communityActive: false,
    studiosActive: false,
    eventsActive: true
  });
}


renderCommunityOrStudios = () => {
  if (this.state.studiosActive === true){
    return <Studios />;
  } else if (this.state.eventsActive === true) {
    return <div>
      <div className='selected-event'>
        <SelectedEvent />
      </div>
      <div className='main-events'> 
        <Events /> </div>
    </div>;
  } else {
    return <Community />; 
  }
}

suggestLocation = () => {
  const suggestions = this.state.locations.map((suggestion, index) => {
    return (
      <option 
        onClick = {() => this.selectedLocation()}
        key={index} value={suggestion}/>
    );
  });
  return suggestions;
}


render () {
  

  const city = this.props.selectedLocation;
  const selectedLocation = city.length ? city : 'USA';
  const singinOut =  this.props.user.username ? <div className="txt" onClick={() => userSignout()}>Log Out</div> : <div className="txt">Sign Up/ SignIn</div>;

  return (
    <div className='body'>
      <div className="nav">
        <div className="logo">
          <div className="logocir">AWC</div>
        </div>

        <form className='location-form'>
          <input
            type='text'
            list='locations'
            onInput={(e) => this.updateState(e)}
            placeholder='Start typing your City'
            className='city-input'
          />
          <button 
            className='submit-location'
            onClick = {(e) => this.selectedLocation(e)}>Search</button>
        </form>
        <datalist 
          id='locations'>
          {this.suggestLocation()}
        </datalist>
        <div className='actual-location'>
          <i className="material-icons">location_on</i>
          <h1>{selectedLocation}</h1>
        </div>

        <div className="buffer"></div>
        <NavLink to='/signin' className="signink">
          <div className="txt">{singinOut}</div>
          <div className="ico">
            <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
              <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"/>
              <path d="M0 0h24v24H0z" fill="none"/>
            </svg>
          </div>
        </NavLink>
      </div>
      <div className="main">
        <div className="profile box">
          <NavBar makeCommunityActive={this.makeCommunityActive}
            makeEventsActive={this.makeEventsActive}
            makeStudioActive={this.makeStudioActive}
          />
        </div>
        <div className="posts box">
          <div className="filters">
            <NavLink to='/' className='main-btn' onClick={() => this.makeCommunityActive()}>Post!</NavLink>
            <NavLink to='/entertainment' className='main-btn'>Videos</NavLink>
            <button className='main-btn' onClick={() => this.makeEventsActive()}>Events</button>
            <button className='main-btn' onClick={() => this.makeStudioActive()}>Studios</button>
          </div>
          <div className="main-box">
            {this.renderCommunityOrStudios()}
          </div> 
        </div>
        <div className="stack">
          <div id="studios" className="stacked  box">
          </div>
          <div id="featured" className="stacked  box">
            <h3>Featured Video</h3>
            <iframe width="350" height="215" src="https://www.youtube.com/embed/2X2tCidTfEU?rel=0" frameBorder="0" allow="autoplay; encrypted-media" allowFullScreen></iframe>
          </div>
        </div>
      </div>
    </div>
  );
}
}

export const mapStateToProps = (state) => {
  return ({
    suggestedEvents: state.suggestedEvents,
    selectedLocation: state.location,
    user: state.user
  });
};

export const mapDispatchToProps = dispatch => ({
  addLocation: (location) => dispatch(addLocation(location)),
  addEvents: (eventsData) => dispatch(addEvents(eventsData)),
  addStudios: (studiosData) => dispatch(addStudios(studiosData))  
});

Main.propTypes = {
  suggestedEvents: PropTypes.object,
  addLocation: PropTypes.func,
  addEvents: PropTypes.func,
  addStudios: PropTypes.func,
  selectedLocation: PropTypes.string
};

export default connect(mapStateToProps, mapDispatchToProps)(Main);