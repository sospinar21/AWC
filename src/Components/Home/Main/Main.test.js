import {Main, mapStateToProps, mapDispatchToProps} from '../Main/Main'
import React from 'react';
import ReactDOM from 'react-dom';
import { mount, shallow } from 'enzyme';

describe('main', () => {

  let main

  beforeEach(() => {

    main = mount(<Main /> , { disableLifecycleMethods: true })

  })

  it('matches snapshot', () => {
    expect(main).toMatchSnapshot();
  })
})