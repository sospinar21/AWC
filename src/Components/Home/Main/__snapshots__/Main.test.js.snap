// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main matches snapshot 1`] = `undefined`;
